docs/standard/serialization/system-text-json/polymorphism.md

```diff
@@ -136,7 +139,607 @@ The following example shows the JSON that results from the preceding code:
+```
+
+```vb
+Imports System.Text.Json
+Imports System.Text.Json.Serialization
+
+Module Program
+    Sub Main()
+        PerformRoundTrip(Of BasePoint)()
+        PerformRoundTrip(Of ThreeDimensionalPoint)()
+        PerformRoundTrip(Of FourDimensionalPoint)()
+    End Sub
+
+    Private Sub PerformRoundTrip(Of T As {BasePoint, New})()
+        Dim json = JsonSerializer.Serialize(Of BasePoint)(New T())
+        Console.WriteLine(json)
+
+        Dim result As BasePoint = JsonSerializer.Deserialize(Of BasePoint)(json)
+        Console.WriteLine($"result is {GetType(T)}; // {TypeOf result Is T}")
+        Console.WriteLine()
+    End Sub
+End Module
+' Sample output:
+'   { "X": 649, "Y": 754 }
+'   result is BasePoint; // True
+'
+'   { "$type": 3, "Z": 247, "X": 814, "Y": 56 }
+'   result is ThreeDimensionalPoint; // True
+'
+'   { "$type": "4d", "W": 427, "Z": 193, "X": 112, "Y": 935 }
+'   result is FourDimensionalPoint; // True
+```
+
+## Configure polymorphic serialization
+
+To customize the type discriminator, use the <xref:System.Text.Json.Serialization.JsonPolymorphicAttribute>. The following example shows how to configure the type discriminator name:
+
+```csharp
+[JsonPolymorphic(TypeDiscriminatorPropertyName = "$discriminator")]
+[JsonDerivedType(typeof(ThreeDimensionalPoint), typeDiscriminator: "3d")]
+public class BasePoint
+{
+    public int X { get; set; }
+    public int Y { get; set; }
+}
+
+public sealed class ThreeDimensionalPoint : BasePoint
+{
+    public int Z { get; set; }
+}
+```
+
+```vb
+<JsonPolymorphic(TypeDiscriminatorPropertyName:="$discriminator")>
+<JsonDerivedType(GetType(ThreeDimensionalPoint), "3d")>
+Public Class BasePoint
+    Public Property X As Integer
+    Public Property Y As Integer
+End Class
+
+Public Class ThreeDimensionalPoint
+    Inherits BasePoint
+    Public Property Z As Integer
+End Class
+```
+
+In the preceding code, the `JsonPolymorphic` attribute configures the `TypeDiscriminatorPropertyName` to the `"$discriminator"` value. With the type discriminator name configured, the following example shows the `ThreeDimensionalPoint` type serialized as JSON:
+
+```csharp
+BasePoint point = new ThreeDimensionalPoint { X = 1, Y = 2, Z = 3 };
+var json = JsonSerializer.Serialize<BasePoint>(point);
+Console.WriteLine(json);
+// Sample output:
+//  { "$discriminator": "3d", "X": 1, "Y": 2, "Z": 3 }
+```
+
+```vb
+Dim point As BasePoint = New ThreeDimensionalPoint With { .X = 1, .Y = 2, .Z = 3 }
+Dim json As String = JsonSerializer.Serialize(Of BasePoint)(point)
+Console.WriteLine(json)
+' Sample output:
+'  { "$discriminator": "3d", "X": 1, "Y": 2, "Z": 3 }
+```
+
+> [!TIP]
+> To avoid potential JSON naming collisions where the <xref:System.Text.Json.Serialization.JsonPolymorphicAttribute.TypeDiscriminatorPropertyName?displayProperty=nameWithType> is errantly set to a name that already exists on the type hierarchy, the type discriminator name should be prefixed with a dollar sign (`$`).
```

Comment:
It's technically possible to have regular properties with a $ prefix, ($ has no special meaning in JSON), so I'd probably phrase this as general advice along the lines of "don't use a type discriminator if it conflicts with a property in your hierarcy"

Reply:
Fixed